### ISSUE TITLE
bug: binary operator not detected as such when preceded by block comment

### ISSUE THREAD

**Issue by Chris00** [tree-sitter/tree-sitter-rust#207]
### Did you check existing issues?

- [ ] I have read all the [tree-sitter docs](https://tree-sitter.github.io/tree-sitter/using-parsers) if it relates to using the parser
- [X] I have searched the existing issues of tree-sitter-rust

### Describe the bug

Pasing `const X: u32 = 1 /* comment */ + 1;` considers the comment as the binary operator (see first parse tree below) while it should be basically parsed as `const X: u32 = (1 /* comment */) + 1;` (see second parse tree below).

### Steps To Reproduce/Bad Parse Tree

```
(source_file
 (const_item const name: (identifier) : type: (primitive_type) =
  value: (binary_expression left: (integer_literal) operator: (block_comment)
			    right: +
	  (integer_literal))
  ;))
```
### Expected Behavior/Parse Tree
```
(source_file
 (const_item const name: (identifier) : type: (primitive_type) =
  value: 
   (binary_expression
    left: (parenthesized_expression ( (integer_literal) (block_comment) ))
    operator: + right: (integer_literal))
  ;))
```
### Repro

```Rust
const X: u32 = 1 /* comment */ + 1;
```

**Comment by amaanq** [comment]
Hm not sure i see the same on my end, this is what I'm seeing:

```
(source_file [0, 0] - [1, 0]
  (const_item [0, 0] - [0, 35]
    name: (identifier [0, 6] - [0, 7])
    type: (primitive_type [0, 9] - [0, 12])
    value: (binary_expression [0, 15] - [0, 34]
      left: (integer_literal [0, 15] - [0, 16])
      (block_comment [0, 17] - [0, 30])
      right: (integer_literal [0, 33] - [0, 34]))))
```

Which at first glance, might look incorrect because the operator is missing. The CLI omits them when they are just literals, looking at it with some more info (on an unreleased fork or neovim's InspectTree) results in this:

![image](https://github.com/tree-sitter/tree-sitter-rust/assets/29718261/a251a708-05ac-4890-8a22-06f78857eaf5)
![image](https://github.com/tree-sitter/tree-sitter-rust/assets/29718261/14ca416d-4ad6-4590-a5b4-58af4a3fbb5a)

Which seems ok to me. Let me know if I'm wrong!
